service/matching/physical_task_queue_manager.go

```diff
@@ -540,6 +527,79 @@ func (c *physicalTaskQueueManagerImpl) TrySyncMatch(ctx context.Context, task *i
 	return c.matcher.Offer(childCtx, task)
 }
 
+func (c *physicalTaskQueueManagerImpl) ensureRegisteredInDeployment(
+	ctx context.Context,
+	namespaceEntry *namespace.Namespace,
+	pollMetadata *pollMetadata,
+) error {
+	if !pollMetadata.workerVersionCapabilities.UseVersioning {
```

Comment:
You can use `worker_versioning.DeploymentFromCapabilities` to get the deployment if v3 is used. Note that we don't want to register if v1-2 is used.